### PR TITLE
bugfix:S3C-2571 enable accept ranges header

### PR DESCRIPTION
**Pull request by ilkescality**
HeadObject supports "Accept-ranges" header for request with `Range` (String | i.e. "bytes=0-9") param.

### CODE REVIEW

File: lib/utilities/collectResponseHeaders.js [hunk 1]
```diff
@@ -45,6 +45,9 @@ function collectResponseHeaders(objectMD, corsHeaders, versioningCfg,
         responseMetaHeaders['x-amz-server-side-encryption-aws-kms-key-id']
             = objectMD['x-amz-server-side-encryption-aws-kms-key-id'];
     }
+    if (objectMD['accept-ranges']) {
```
Comment (jonathan-gramain) [inline lib/utilities/collectResponseHeaders.js]: Why is this part of `objectMD`? For me it's a server-side capability that is not related to object internal metadata.

Comment (ilkescality) [reply lib/utilities/collectResponseHeaders.js]: I thought so, but wasn't sure, I tried to leave a comment on my original PR to check, removing it then.

File: tests/unit/api/objectHead.js [hunk 2]
```diff
@@ -137,6 +137,27 @@ describe('objectHead API', () => {
         });
     });
 
+    it('should return Accept-Ranges header if request header includes' +
```
Comment (miniscruff) [inline tests/unit/api/objectHead.js]: your code changes introduced a total of 4 new paths through the method but you only added one new test. Not a requirement but consider adding tests for all paths. Just my general practice anyway.

File: lib/api/objectHead.js [hunk 3]
```diff
@@ -77,8 +78,47 @@ function objectHead(authInfo, request, log, callback) {
             if (headerValResult.error) {
                 return callback(headerValResult.error, corsHeaders);
             }
+            const responseHeaders = collectResponseHeaders(objMD, corsHeaders,
```
Comment (miniscruff) [inline lib/api/objectHead.js:81]: This method is quite long now ( roughly 80 lines ) and although eslint doesn't require it I would suggest splitting it up if possible. I generally recommend no more than about a "screen" tall. In practice this is varied but 20-40 lines is a good range in most cases for me but would, of course, adjust based on language or needs.

File: tests/unit/api/objectHead.js (hunk 2 above)
Comment (ilkescality) [reply tests/unit/api/objectHead.js]: done; added new unit tests and all the paths should be covered now

File: lib/api/objectHead.js (hunk 3 above)
Comment (ilkescality) [reply lib/api/objectHead.js:81]: I agree about this; for now, I copied the logic from the objectGet API as it is, in order to to keep those consistent and not to impact the scope of this PR.
I plan to create a follow-up ticket to submit a refactor PR to address your comment as well as some other closely related issues that I encountered.

**Review by tmacro — APPROVED** [review]

File: lib/api/objectHead.js [hunk 4]
```diff
@@ -87,10 +127,8 @@ function objectHead(authInfo, request, log, callback) {
                     return callback(errors.InvalidRange, corsHeaders);
                 }
                 // eslint-disable-next-line no-param-reassign
-                objMD['content-length'] = partSize;
+                responseHeaders['content-length'] = partSize;
```
Comment (dora-korpar) [inline lib/api/objectHead.js:129]: minor, but you I don't think you need the eslint comment above this line anymore

Comment (ilkescality) [reply lib/api/objectHead.js:129]: good catch

**Review by nicolas2bert — APPROVED** [review]

**Comment by ilkescality** [comment]
@bert-e reset

**Comment by ilkescality** [comment]
@bert-e force_reset

**Comment by ilkescality** [comment]
@bert-e approve

**Review by rahulreddy — APPROVED** [review]
